src/commonMain/kotlin/fr/acinq/lightning/db/PaymentsDb.kt

```diff
@@ -142,13 +139,17 @@ data class IncomingPayment(val preimage: ByteVector32, val origin: Origin, val r
         /** KeySend payments are spontaneous donations for which we didn't create an invoice. */
         object KeySend : Origin()
 
-        /** Swap-in works by sending an on-chain transaction to a swap server, which will pay us in exchange. We may not know the origin address. */
+        /** DEPRECATED: this is the legacy trusted swap-in, which we keep for backwards-compatibility (previous payments inside the DB). */
         data class SwapIn(val address: String?) : Origin()
 
+        /** Trust-less swap-in based on dual funding. */
+        data class DualSwapIn(val localInputs: Set<OutPoint>) : Origin()
```

Comment:
@dpad85 let me know if that works for you? I believe this lets us point to explorers that will show the swap transactions.

Reply:
Yep it's better to have the outpoint/transaction than just the address as we used to with legacy swap-in.